lib/phoenix_live_view/channel.ex

```diff
@@ -1251,10 +1251,10 @@ defmodule Phoenix.LiveView.Channel do
     end)
   end
 
-  defp writer!(socket, writer) do
+  defp writer!(socket, name, entry, writer) do
     result =
       case writer do
-        writer when is_function(writer, 1) -> writer.(socket)
+        writer when is_function(writer, 3) -> writer.(name, entry, socket)
         other -> other
```

Comment:
let's raise. We can simplify and always require the closure

Reply:
Ready for another look! :)